app/views/admin/api/edit.html.erb

```diff
@@ -0,0 +1,18 @@
+<%= form_for @api_key, url: admin_api_path(@api_key) do |f| %>

```

Comment:
This needs our standard page structure -- i.e. the admin div, a heading, and navigation. It also has some invalid HTML (a `<dl>` can only have `<dt>` and `<dd>` as its direct children, but this has a `<ul>`) and isn't quite set up like our usual forms, which are grouped into field sets with legends and headings that repeat the legend text (which is not the most efficient thing in the world, but screen readers don't always get along with the `<legend>` element).

I noticed this form, unlike the form for creating a new API token, doesn't require a name. Does that mean you can use this form to remove the token's name, save, and everything will work properly when the API token doesn't have a name? If so, awesome! If not, though, the Name field should be marked as required and the required notice should be included at the top of the form.

Here's how it should be set up, assuming the field isn't required:

```
<div class="admin">
  <!--Descriptive page name, messages and instructions-->
  <h2 class="heading"><%= ts("Edit API Token") %></h2>
  <!--/descriptions-->

  <!--subnav-->
  <ul class="navigation actions" role="navigation">
    <li><%= span_if_current ts("API Tokens"), admin_api_path %></li>
    <li><%= span_if_current ts("New API Token"), new_admin_api_path %></li>
    <li><%= span_if_current ts("Delete API Token"), admin_api_path(@api_key), :confirm => ts("Are you sure you want to delete this API key?"), :method => :delete %></li>
  </ul>
  <!--/subnav-->

  <!--main content-->
  <%= form_for @api_key, url: admin_api_path(@api_key), html: { class: "post" } do |f| %>
    <fieldset>
      <legend><%= ts("API Token") %></legend>
      <h3 class="landmark heading"><%= ts("API Token") %></h3>
      <dl>
        <dt><%= f.label :name, ts("Name") %></dt>
        <dd>
          <%= f.text_field :name %>
          <%= live_validation_for_field("api_key_name", :presence => false, :maximum_length => Profile::PROFILE_TITLE_MAX) %>
        </dd>
        <dt>
          <%= f.label :access_token, ts("Token (automatically generated)") %>
        </dt>
        <dd>
          <%= f.text_field :access_token, readonly: true %>
        </dd>
      </dl>
    </fieldset>
    <fieldset>
      <legend><%= ts("Actions") %></legend>
      <h3 class="landmark heading"><%= ts("Actions") %></h3>
      <p class="actions">
        <%= f.submit ts('Update API Token') %>
      </p>
    </fieldset>
  <% end %>
  <!--/content-->
</div>
```

Since a lot of this is shared between multiple files, it may be worth considering breaking the form and navigation out into partials and just using logic to tweak where needed, like [with the locales section here](https://github.com/zz9pzza/otwarchive/commit/1c09f7e97140a237271e3968fcefaeaa6f8eb147) or [the banners section](https://github.com/otwcode/otwarchive/tree/master/app/views/admin/banners). But if you'd rather not fool with it, not a problem!